localstack/services/sns/provider.py

```diff
@@ -1102,7 +1121,20 @@ async def message_to_subscriber(
     elif subscriber["Protocol"] == "application":
         try:
             sns_client = aws_stack.connect_to_service("sns")
-            sns_client.publish(TargetArn=subscriber["Endpoint"], Message=message)
+            if (msg_structure := req_data.get("MessageStructure")) and msg_structure[0] == "json":
+                # only valid value for MessageStructure is json, we cannot set it to nothing
+                sns_client.publish(
+                    TargetArn=subscriber["Endpoint"],
+                    Message=message,
+                    MessageAttributes=message_attributes,
+                    MessageStructure="json",
+                )
+            else:
+                sns_client.publish(
+                    TargetArn=subscriber["Endpoint"],
+                    Message=message,
+                    MessageAttributes=message_attributes,
+                )
```

Comment:
I am a bit confused by the comment and the 2 "same but different" method calls. I would have assumed that this means MessageAttributes either needs to be "json", or not be present at all. But there is no consequence if the value is not json, and I assume it will still be treated as json if MessageStructure is not present. Is that true?
If it is true, then we could simply always set MessageStructure = "json", or always omit it, purely semantically speaking.
If AWS simply behaves this way (setting it implicitly or explicitly as json), you could potentially do something like this (pure nitpick, this is probably a matter of preference).
```
publish_args = {
    TargetArn=subscriber["Endpoint"],
    Message=message,
    MessageAttributes=message_attributes,
}
if (msg_structure := req_data.get("MessageStructure")) and msg_structure[0] == "json":
    publish_args.update({MessageStructure': 'json'})
    
sns.publish(**publish_args)
```
Whether this is easier to read or not is highly debatable, but it highlights that this is the same call in a "different flavor".
IF that is the case, as stated above.

Reply:
Right, it's exactly what you said. At least that's the behaviour that AWS is showing on the docs: you either set it to `"json"`, or you don't set it. It's not on by default, it's "nothing" if not set, there's not "default" format, it's a string one. The only valid value is "json".
But the kwargs is much, much cleaner. I'll change to this, thanks a lot for this nit.